specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/preview/2021-02-01-preview/iotdps.json

```diff
@@ -0,0 +1,2622 @@
+            "type": "string",
+            "description": "Name of the certificate to retrieve."
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Resource group identifier."
+          },
+          {
+            "name": "provisioningServiceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Name of the provisioning service the certificate is associated with."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "ETag of the certificate."
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Metadata for the specified certificate.",
+            "schema": {
+              "$ref": "#/definitions/CertificateResponse"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PUT"
+        ],
+        "summary": "Upload the certificate to the provisioning service.",
+        "description": "Add new certificate or update an existing certificate.",
+        "operationId": "DpsCertificate_CreateOrUpdate",
+        "x-ms-examples": {
+          "DPSCreateOrUpdateCertificate": {
+            "$ref": "./examples/DPSCertificateCreateOrUpdate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "Resource group identifier.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "provisioningServiceName",
+            "in": "path",
+            "description": "The name of the provisioning service.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "certificateName",
+            "in": "path",
+            "description": "The name of the certificate create or update.",
+            "required": true,
+            "type": "string",
+            "maxLength": 256
+          },
+          {
+            "name": "certificateDescription",
+            "in": "body",
+            "description": "The certificate body.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CertificateBodyDescription"
```

Comment:
Please use the same definition for both the request body and response. You can mark properties in the response only as read-only.

Reply:
After discussion with @j5lim in the ARM review meeting on Jan 14. We agreed to temporarily resolves these conversations as they are not related to the changes introduced in this 2021-02-01-preview version. Also dps team needs to prioritize this new preview API and deliver it in a timely manner

Comments in this PR are captured in these 2 items and will get fixed in the next swagger PR in the future.
Product Backlog Item [9134544](https://msazure.visualstudio.com/One/_workitems/edit/9134544): Resolve Comments in RP 20210201Preview Swagger PR
Product Backlog Item [9134490](https://msazure.visualstudio.com/One/_workitems/edit/9134490): Add systemData Property in RP Swagger